tests/Api/Shop/OrdersTest.php

```diff
@@ -101,28 +121,82 @@ public function it_gets_order_adjustments(): void
+        $response = $this->client->getResponse();
+
+        $this->assertResponseCode($response, Response::HTTP_UNAUTHORIZED);
+    }
+
+    /** @test */
+    public function it_prevents_visitor_from_trying_to_get_the_order_item_adjustments_of_logged_in_user(): void
+    {
+        $this->loadFixturesFromFiles([
+            'channel.yaml',
+            'cart.yaml',
+            'country.yaml',
+            'authentication/customer.yaml',
+            'shipping_method.yaml',
+            'payment_method.yaml',
+        ]);
 
         $tokenValue = $this->pickUpCart();
 
-        $this->addItemToCart('MUG_BLUE', 3, $tokenValue);
+        $order = $this->createOrderWithOrderItemAdjustments($tokenValue, 'oliver@doe.com');
 
-        /** @var OrderInterface $order */
-        $order = $this->get('sylius.repository.order')->findCartByTokenValue($tokenValue);
-        $orderItem = $order->getItems()->first();
+        $this->client->request('GET', '/api/v2/shop/orders/nAWw2jewpA/items/'.$order->getItems()->first()->getId().'/adjustments', [], [], self::CONTENT_TYPE_HEADER);
+        $response = $this->client->getResponse();
 
-        /** @var AdjustmentInterface $adjustment */
-        $adjustment = $this->get('sylius.factory.adjustment')->createNew();
+        $this->assertResponseCode($response, Response::HTTP_UNAUTHORIZED);
+    }
 
-        $adjustment->setType(AdjustmentInterface::ORDER_ITEM_PROMOTION_ADJUSTMENT);
-        $adjustment->setAmount(200);
-        $adjustment->setNeutral(false);
-        $adjustment->setLabel('Test Promotion Adjustment');
+    public function it_returns_nothing_if_logged_in_shop_user_tries_to_get_the_order_item_adjustments_of_another_user(): void
+    {
+        $this->loadFixturesFromFiles([
+            'channel.yaml',
+            'cart.yaml',
+            'country.yaml',
+            'authentication/customer.yaml',
+            'shipping_method.yaml',
+            'payment_method.yaml',
+        ]);
+
+        $loginData = $this->logInShopUser('dave@doe.com');
+        $authorizationHeader = self::$kernel->getContainer()->getParameter('sylius.api.authorization_header');
+        $header['HTTP_' . $authorizationHeader] = 'Bearer ' . $loginData;
+        $header = array_merge($header, self::CONTENT_TYPE_HEADER);
 
-        $orderItem->addAdjustment($adjustment);
-        $this->get('sylius.manager.order')->flush();
+        $tokenValue = $this->pickUpCart();
+
+        $order = $this->createOrderWithOrderItemAdjustments($tokenValue, 'oliver@doe.com');
+
+        $this->client->request('GET', '/api/v2/shop/orders/nAWw2jewpA/items/'.$order->getItems()->first()->getId().'/adjustments', [], [], $header);
+        $response = $this->client->getResponse();
+
+        $this->assertResponseCode($response, Response::HTTP_UNAUTHORIZED);
+    }
+
+    /** @test */
+    public function it_gets_order_item_adjustments(): void
```

Comment:
Since we already have a split in the actions between the user/visitor, maybe we should also duplicate all the other tests in this file per case.